org.adempiere.base/src/org/compiere/print/ReportEngine.java

```diff
@@ -992,7 +992,23 @@ else if (obj instanceof PrintDataElement)
 											}
 											if (tableID > 0) {
 												MTable mTable = MTable.get(getCtx(), tableID);
-												String foreignColumnName = mTable.getTableName() + "_ID";
+												String tableName = mTable.getTableName();
+												
+												ArrayList<MColumn> list = new ArrayList<MColumn>();
+												for (String idColumnName : mTable.getIdentifierColumns()) {
+													MColumn column = mTable.getColumn(idColumnName);
+													list.add (column);
+												}
+												if(list.size() > 0) {
+													StringBuilder displayColumn = new StringBuilder(list.get(0).getColumnName());
```

Comment:
We should use value from MSysConfig.IDENTIFIER_SEPARATOR to combine the list of value from identifiers (use nvl to handle null). I've not tested but the ColumnLookup.getLookupDisplay approach might work here as well.